Traceback (most recent call last):
  File "/home/jeb/Projects/training/regularization-using-data/train.py", line 171, in <module>
    out, accuracies_mean = jax.lax.scan(
                           ^^^^^^^^^^^^^
  File "/home/jeb/Projects/training/regularization-using-data/train.py", line 143, in training
    train_state, eval_metrics = jax.lax.cond(
                                ^^^^^^^^^^^^^
TypeError: cond branch outputs must have the same pytree structure, but they differ:

true_fun is <lambda> at /home/jeb/Projects/training/regularization-using-data/train.py:145
false_fun is <lambda> at /home/jeb/Projects/training/regularization-using-data/train.py:146

true_fun output at path [1] is a <class 'training.training.EvalMetrics'> but false_fun output at path [1] is a <class '__main__.EvalMetrics'>, so their Python types differ.

Revise true_fun and/or false_fun so that they have the same pytree structure.
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
